### ISSUE TITLE
Sort order in the file browser should be remembered in user's browser's localstorage

### ISSUE THREAD

**Issue by andmis** [jupyterlab/jupyterlab#15414]
### Problem

I have a lot of notebook files and they are sorted by name in a meaningful way. The sort order in the file browser is important for my workflow.

Unfortunately, if I close my browser tab and then open a new tab to my Jupyter Lab session, the sort order has reset to its default.

### Proposed Solution

Memorize the sort order in the user's browser's localstorage.

**Comment by krassowski** [comment]
Yes please. This could technically be implemented as either layout restoration, or a setting. Visible columns are a setting, but I think the sort state makes more sense to be stored in the state DB.

One of the relevant places in the codebase would be:

https://github.com/jupyterlab/jupyterlab/blob/7d7d57a9e7b4a9f16189f7d79875b2c773243c0a/packages/filebrowser/src/model.ts#L326-L382

you can see how the path is saved here:

https://github.com/jupyterlab/jupyterlab/blob/7d7d57a9e7b4a9f16189f7d79875b2c773243c0a/packages/filebrowser/src/model.ts#L274-L278

Another question would be whether we want tostore the sort state per directory or globally (some filebrowsers store it per directory).
